integration_tests/test_builtin_len.py

```diff
@@ -6,5 +6,13 @@ def test_len():
     assert len(s) == 0
     assert len("abcd") == 4
     assert len("") == 0
+    t: str
+    t = "efg"
+    s = "abcd" + t
+    i: i32
+    i = len(s)
+    assert i == 7
+    i = len("abc" + "def")
+    assert i == 6
```

Comment:
Add a test:
```
assert len(t + s) == 10
```
It seems based on the code that the result will be:
```
assert t + s.size() == 0;
```
But rather it should be:
```
assert (t+s).size() == 0;
```
```

(We also have to implement the `assert` node correctly, probably using a C++ macro, but that's a separate issue.)